modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/LiferayExperienceCloud/components/SetupLXCModal/components/SetupLXCForm/components/SetupLXCPage/index.jsx

```diff
@@ -85,116 +146,119 @@ const SetupLiferayExperienceCloudPage = ({
 									/>
-								))}
+
+									<Select
+										groupStyle="mb-0"
+										key={primaryRegionList}
+										label={i18n.translate('primary-region')}
+										name="lxc.primaryRegion"
+										options={primaryRegionList}
+										required
+									/>
+								</ClayForm.Group>
+
+								<ClayForm.Group className="mb-0">
+									{values.lxc.admins.map((admin, index) => (
+										<AdminInputs
+											admin={admin}
+											id={index}
+											key={index}
+										/>
+									))}
+								</ClayForm.Group>
 							</ClayForm.Group>
-						</ClayForm.Group>
 
-						{values?.lxc?.admins?.length >
-							INITIAL_SETUP_ADMIN_COUNT && (
+							{values?.lxc?.admins?.length >
+								INITIAL_SETUP_ADMIN_COUNT && (
+								<Button
+									className="ml-3 my-2 text-brandy-secondary"
+									displayType="secondary"
+									onClick={() => {
+										pop();
+										setBaseButtonDisabled(false);
+									}}
+									prependIcon="hr"
+									small
+								>
+									{i18n.translate('remove-project-admin')}
+								</Button>
+							)}
+
 							<Button
-								className="ml-3 my-2 text-brandy-secondary"
-								displayType="secondary"
+								className="cp-btn-add-dxp-cloud ml-3 my-2 rounded-xs"
 								onClick={() => {
-									pop();
-									setBaseButtonDisabled(false);
+									push(
+										getInitialLxcAdmins(values?.lxc?.admins)
+									);
+									setBaseButtonDisabled(true);
 								}}
-								prependIcon="hr"
+								prependIcon="plus"
 								small
 							>
-								{i18n.translate('remove-project-admin')}
+								{i18n.translate('add-another-admin')}
 							</Button>
-						)}
-
-						<Button
-							className="cp-btn-add-dxp-cloud ml-3 my-2 rounded-xs"
-							onClick={() => {
-								push(getInitialLxcAdmins(values?.lxc?.admins));
-								setBaseButtonDisabled(true);
-							}}
-							prependIcon="plus"
-							small
-						>
-							{i18n.translate('add-another-admin')}
-						</Button>
-
-						<hr />
-
-						<ClayForm.Group className="mb-0">
-							<Input
-								groupStyle="pb-1"
-								label={i18n.translate(
-									'incident-management-contacts-first-and-last-name'
-								)}
-								name="lxc.incidentManagementFullName"
-								required
-								type="text"
-							/>
-
-							<Input
-								groupStyle="pb-1"
-								helper={i18n.translate(
-									'lowercase-letters-and-numbers-only-project-ids-cannot-be-changed'
-								)}
-								label={i18n.translate(
-									'incident-management-contacts-email-address'
-								)}
-								name="lxc.incidentManagementEmail"
-								required
-								type="text"
-								validations={[
-									(value) =>
-										isValidEmail(value, bannedDomains),
-								]}
-							/>
-						</ClayForm.Group>
-					</>
-				)}
-			/>
+
+							<hr />
+						</>
+					)}
+				/>
+			)}
+
+			{step === 2 && (
+				<div>
+					<SetupHighPriorityContactForm
+						addContactList={addHighPriorityContacts}
```

Comment:
Try to use an 'enum' or a object to identify each step here and in the other steps for better organization if possible.